pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -178,12 +181,40 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 	// If the usersignup is not banned and not deactivated then ensure the deactivated notification status is set to false.
 	// This is especially important for cases when a user is deactivated and then reactivated because the status is used to
 	// trigger sending of the notification. If a user is reactivated a notification should be sent to the user again.
+
 	if !banned && !userSignup.Spec.Deactivated {
 		if err := r.updateStatus(logger, userSignup, r.setStatusDeactivationNotificationUserIsActive); err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
+	// If the usersignup is not banned and not within the pre-deactivation period then ensure the deactivation notification
+	// status is set to false. This is especially important for cases when a user is deactivated and then reactivated
+	// because the status is used to trigger sending of the notification. If a user is reactivated a notification should
+	// be sent to the user again.
+	if !banned && !states.Deactivating(userSignup) {
+		if err := r.updateStatus(logger, userSignup, r.setStatusDeactivatingNotificationNotInPreDeactivation); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	if states.Deactivating(userSignup) && condition.IsNotTrue(userSignup.Status.Conditions,
+		toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated) {
```

Comment:
Based on our experience with the outdated values in the `status` part of the `UserSignup` resource I'm wondering if this is a safe way to do - to decide what to do based on the value in `UserSignup.status`.
Anyway, I don't think that it's so critical because:
* we saw the issue with the outdated values during provisioning when there are a lot of changes in both resources `UserSignup` and `MUR` which triggers many reconciles. But this is not the case - there should be so many changes, so I believe that we should not get the outdated value so often.
* in the worst case, we will send the notification twice, which is not a big deal :-)

Reply:
This is the same flow that we use for deactivation also, further down in the code.  I'm concerned about the outdated resource issue too, and I think we should look at our general methodology for how we update resources within our controllers and possibly refactor to reduce the number of distinct updates.  However for now I don't have an easy fix for this, and as you say in the worst case we will send the notification twice.

Reply:
Yeah, I agree with you. 
My guess is that we should not rely on the values set in status part inside the controller - to decide what to do next. Maybe, we should rather use annotations instead.